partition_manager: Update documentation (#8470)
nrfconnect/sdk-nrf: scripts/partition_manager/partition_manager.rst

Any placeholder partition can specify that it should be stored in the external flash memory region.
External flash memory regions always use the *start_to_end* placement strategy.

> To store partitions in the external flash memory, you can choose a value for the ``nordic,pm-ext-flash`` property in the devicetree,
> or directly use external flash DTS node label as DEVICE.

Reviewer: ```suggestion
To store partitions in the external flash memory, you can either choose a value for the ``nordic,pm-ext-flash`` property in the devicetree, or directly use an external flash DTS node label as ``DEVICE``.
```

Is the 2nd part of the sentence supposed to mean that the " external flash DTS node label" should be used as "DEVICE" or that the  external flash DTS node should be labelled as "DEVICE"?

Author: The former.

.. _pm_build_system:

> You must enable device drivers for each flash partition.
> To do so, set the ``DEFAULT_DRIVER_KCONFIG`` parameter to the Kconfig option that enables the driver during compilation.
> When a driver for a partition is not enabled, such partition will not be included into partition manager and will not be accessible at runtime.
> You can use :kconfig:option:`CONFIG_PM_OVERRIDE_EXTERNAL_DRIVER_CHECK` to override the behaviour and prevent compile-time check for external flash memory devices.
> This is useful when the external device does not have a driver controlled by kconfig options.

Reviewer 2: ```suggestion
You must enable device drivers for each flash partition.
Set the ``DEFAULT_DRIVER_KCONFIG`` parameter to the Kconfig option that enables the driver during compilation.
When a driver for a partition is not enabled, the partition will not be included in the partition manager and will not be accessible at runtime.
You can use the :kconfig:option:`CONFIG_PM_OVERRIDE_EXTERNAL_DRIVER_CHECK` Kconfig option to override the behavior and prevent compile-time check for external flash memory devices.
This is useful when the external device does not have a driver controlled by Kconfig options.
```

.. _pm_build_system:

You must enable device drivers for each flash partition.
To do so, set the ``DEFAULT_DRIVER_KCONFIG`` parameter to the Kconfig option that enables the driver during compilation.

> When a driver for a partition is not enabled, such partition will not be included into partition manager and will not be accessible at run-time.

Reviewer: ```suggestion
When a driver for a partition is not enabled, such partition will not be included into partition manager and will not be accessible at runtime.
```

> You can use kconfig:option:`CONFIG_PM_OVERRIDE_EXTERNAL_DRIVER_CHECK` to override the behaviour
> and prevent compile-time check for external flash memory devices.

Reviewer: ```suggestion
You can use :kconfig:option:`CONFIG_PM_OVERRIDE_EXTERNAL_DRIVER_CHECK` to override the behaviour and prevent compile-time check for external flash memory devices.
```

> This is useful when the external device does not have a Kconfig option controlled driver.

Reviewer: ```suggestion
This is useful when the external device does not have a driver controlled by kconfig options.
```

.. _pm_build_system:

> Flash partitions requre device drivers to be enabled for them. The ``DEFAULT_DRIVER_KCONFIG``
> parameter should be set to a Kconfig option that enables the driver during compilation.

Reviewer: ```suggestion
You must enable device drivers for each flash partition. 
To do so, set the ``DEFAULT_DRIVER_KCONFIG`` parameter to the Kconfig option that enables the driver during compilation.
```

Could we be more explicit on indicating here which kconfig option enables the driver during compilation?

> When a driver for a partition is not enabled, such partition will not be included into
> partition man and will not be accessible at run-time.

Reviewer: ```suggestion
When a driver for a partition is not enabled, such partition will not be included into partition manager and will not be accessible at run-time.
```

> kconfig:option:`CONFIG_PM_OVERRIDE_EXTERNAL_DRIVER_CHECK` can be used to override the behaviour
> and prevent compile-time check for external flash devices; this is useful when external device
> does not have Kconfig conroller driver.

Reviewer: ```suggestion
You can use kconfig:option:`CONFIG_PM_OVERRIDE_EXTERNAL_DRIVER_CHECK` to override the behaviour
and prevent compile-time check for external flash memory devices.
This is useful when the external device does not have a Kconfig controller driver.
```
